python/ray/_raylet.pyx

```diff
@@ -935,6 +935,34 @@ cdef class CoreWorker:
 
         return RayObjectsToDataMetadataPairs(results)
 
+    def kv_put(self, c_string key, c_string value):
```

Comment:
I think we need a flag that indicates whether the data is for the current job (e.g., serialized functions). For these data, we need to clean them up when the job finishes.

Reply:
The current implementation which we want to get rid of is used to manage global resources, so that's why it doesn't have scope data.

We can add them based on this PR later.